crypto/evp/exchange.c

```diff
@@ -427,11 +427,11 @@ int EVP_PKEY_derive(EVP_PKEY_CTX *ctx, unsigned char *key, size_t *pkeylen)
         goto legacy;
 
     ret = ctx->op.kex.exchange->derive(ctx->op.kex.exchprovctx, key, pkeylen,
-                                       SIZE_MAX);
+                                       key != NULL ? *pkeylen : SIZE_MAX);
```

Comment:
If `key == NULL` shouldn't we be setting this value to 0? This is the size of the buffer, which is obviously 0 if the buffer is NULL.

Reply:
Makes sense, I'll review the provider implementations if they DTRT in that case. That they do really return the real output len in *pkeylen.

Reply:
Done